### PR TITLE
Update maven wrapper to 3.3.2

### PR DESCRIPTION
**Pull request by alex-martel**
This updates the maven wrapper from 3.2.0 to 3.3.2.

~~The new version uses, by default, a script-only variant that does not need the jar, nor the maven downloader java source file. Type can be changed if needed. More info here; https://maven.apache.org/wrapper/index.html#usage-with-or-without-binary-jar~~

### CODE REVIEW

File: .mvn/wrapper/maven-wrapper.properties [hunk 1]
```diff
@@ -14,6 +14,7 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+wrapperVersion=3.3.2
+distributionType=only-script
```
Comment (gsmet) [inline .mvn/wrapper/maven-wrapper.properties:18]: I think I would be more comfortable using the previous approach. A bit worried about odd proxy config on enterprise laptops.
Probably a good idea to let it bake for a while.

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks!

I added a small comment.
As for the failing tests: one you're happy with the changes, you need to do a quick build to install the artifacts and then run `mvn clean verify -Dsnap` in the modules with failing tests (and only them!).

File: .mvn/wrapper/maven-wrapper.properties (hunk 1 above)
Comment (manofthepeace) [reply .mvn/wrapper/maven-wrapper.properties]: I have restored the previous behaviour. i.e. using the binary for the root project, and using the "source" for the devtools.

**Comment by manofthepeace** [comment]
@gsmet  looks like there was a OOM on windows;

```
2024-06-18T18:39:57.9988343Z java.lang.OutOfMemoryError: Java heap space
2024-06-18T18:39:57.9989333Z     at java.util.concurrent.ConcurrentHashMap.putVal (ConcurrentHashMap.java:1019)
2024-06-18T18:39:57.9990905Z     at java.util.concurrent.ConcurrentHashMap.put (ConcurrentHashMap.java:1006)
2024-06-18T18:39:57.9992017Z     at java.util.Collections$SetFromMap.add (Collections.java:5685)
2024-06-18T18:39:57.9993231Z     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1492)
2024-06-18T18:39:57.9994309Z     at java.lang.Thread.run (Thread.java:840)
2024-06-18T18:39:57.9995086Z     at org.jboss.threads.JBossThread.run (JBossThread.java:483)
2024-06-18T18:39:57.9996384Z [ERROR] [org.jboss.threads.errors] Thread Thread[build-47,5,build group] threw an uncaught exception
2024-06-18T18:39:57.9997470Z java.lang.OutOfMemoryError: Java heap space
2024-06-18T18:39:57.9998420Z     at java.util.concurrent.ConcurrentHashMap.putVal (ConcurrentHashMap.java:1019)
2024-06-18T18:39:57.9999695Z     at java.util.concurrent.ConcurrentHashMap.put (ConcurrentHashMap.java:1006)
2024-06-18T18:39:58.0001157Z     at java.util.Collections$SetFromMap.add (Collections.java:5685)
2024-06-18T18:39:58.0002426Z     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1492)
2024-06-18T18:39:58.0003558Z     at java.lang.Thread.run (Thread.java:840)
2024-06-18T18:39:58.0004372Z     at org.jboss.threads.JBossThread.run (JBossThread.java:483)
2024-06-18T18:39:58.0006022Z [ERROR] [org.jboss.threads.errors] Thread Thread[build-30,5,build group] threw an uncaught exception
2024-06-18T18:39:58.0007151Z java.lang.OutOfMemoryError: Java heap space
2024-06-18T19:47:09.4393633Z 2024-06-18 19:47:09
2024-06-18T19:47:09.4396305Z Full thread dump OpenJDK 64-Bit Server VM (17.0.11+9 mixed mode, sharing):
```

**Comment by manofthepeace** [comment]
@gsmet rebased this one, mind taking another look? This does not change the behaviour in how it used to work with the older wrapper; as noted in the comment here ;  https://github.com/quarkusio/quarkus/pull/40911#discussion_r1644466822

**Review by gsmet — APPROVED** [review]

**Pull request merged**